dictionary.go

```diff
@@ -73,7 +73,7 @@ func newByteArrayDictionary(typ Type, columnIndex int16, bufferSize int) *byteAr
 		typ: typ,
 		byteArrayPage: byteArrayPage{
 			values:      encoding.MakeByteArrayList(dictCap(bufferSize, 16)),
-			columnIndex: columnIndex,
+			columnIndex: ^columnIndex,
```

Comment:
I'm confused why we have an XOR here, at least it's not obvious why a struct property with the same name as the variable needs to be modified by a lot.

Reply:
This is a fix for another bug I ran into while debugging the original issue. The reason why we use XOR are documented in https://github.com/segmentio/parquet-go/blob/main/value.go#L39

Technically we could do the XOR later in readers' `ReadValues` method, the alternative here is to compute it ahead of time instead of doing it on each read.

Most of the code was using this approach so I unified on this, we could choose to store the plain value and compute XOR when calling `ReadValues`, it just needs to be consistent across the code base.